plugin/log/log.go

```diff
@@ -26,21 +26,24 @@ type Logger struct {
 // ServeDNS implements the plugin.Handler interface.
 func (l Logger) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
 	state := request.Request{W: w, Req: r}
+	name := state.Name()
 	for _, rule := range l.Rules {
-		if !plugin.Name(rule.NameScope).Matches(state.Name()) {
+		if !plugin.Name(rule.NameScope).Matches(name) {
 			continue
 		}
 
 		rrw := dnstest.NewRecorder(w)
 		rc, err := plugin.NextOrFailure(l.Name(), l.Next, ctx, rrw, r)
 
-		tpe, _ := response.Typify(rrw.Msg, time.Now().UTC())
-		class := response.Classify(tpe)
 		// If we don't set up a class in config, the default "all" will be added
 		// and we shouldn't have an empty rule.Class.
-		_, ok := rule.Class[response.All]
-		_, ok1 := rule.Class[class]
-		if ok || ok1 {
+		_, shouldLog := rule.Class[response.All]
+		if !shouldLog {
+			tpe, _ := response.Typify(rrw.Msg, time.Now().UTC())
+			class := response.Classify(tpe)
+			_, shouldLog = rule.Class[class]
```

Comment:
naming it `shouldLog` doesnt change anything, `shouldLog` is overwritten in this if-statement, that looks very weird.

Reply:
Ok, I thought the confusion was around the variable name itself.  I changed it to back to using `ok` and `ok1`.